lib/hanami/events/subscriber.rb

```diff
@@ -2,12 +2,12 @@ module Hanami
   module Events
     class Subscriber
       def initialize(pattern, block)
-        @pattern = pattern
+        @pattern = Matcher.new(pattern)
         @block = block
       end
 
       def call(event_name, payload)
-        @block.call(payload) if @pattern == event_name
+        @block.call(payload) if @pattern.match?(event_name)
```

Comment:
btw, I've just realized that this `@block` variable could be renamed to more appropriate `event_handler`. Because it can be anything that responds to `#call`. Block is misleading here.